main.tf

```diff
@@ -1,7 +1,7 @@
 locals {
   enabled        = module.this.enabled
   instance_count = local.enabled ? 1 : 0
-  volume_count   = var.ebs_volume_count > 0 && local.instance_count > 0 ? var.ebs_volume_count : 0
+  volume_count   = local.enabled && var.ebs_volume_count > 0 ? var.ebs_volume_count : 0
```

Comment:
why removing `local.instance_coun > 0` ?
Volumes are attached to instances, so if `instance_count = 0` we prob should not try to attach